pre_commit/repository.py

```diff
@@ -190,6 +191,21 @@ def _cloned_repository_hooks(repo_config, store, root_config):
     )
 
 
+_TOP_LEVEL_KEYS = ('repo', 'rev', 'hooks')
+
+
+def _check_extra_top_level_keys(repo_config):
+    extra_top_level_keys = set(dict(repo_config).keys()) - set(_TOP_LEVEL_KEYS)
```

Comment:
there's 4 unnecessary copies happening here:

1. `dict(repo_config)`
2. `.keys()` (in python 2)
3. `set(dict...)`
4. `set(_TOP_LEVEL_KEYS)`

The module constant should be `frozenset((...))`, you can use `set(repo_config)` directly